presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -164,6 +184,7 @@ public void addInput(Page page)
     {
         requireNonNull(page, "page is null");
         checkState(state == State.RUNNING, "Operator is %s", state);
+        checkState(blocked == NOT_BLOCKED, "output is already blocked");
```

Comment:
There is no guarantee that the engine respects the blocked status.  Instead this code should just reset the blocked status to NOT_BLOCKED and executed the code.